rosplan_planning_system/src/ProblemGeneration/ProblemInterface.cpp

```diff
@@ -19,7 +19,18 @@ namespace KCL_rosplan {
 		// connecting to KB
 		std::string kb = "knowledge_base";
 		node_handle->getParam("knowledge_base", kb);
-		pddl_problem_generator.knowledge_base = kb;
+
+		// Check problem fle extension to select a problem generator
+		node_handle->param<std::string>("problem_path", problem_path, "common/problem.pddl");
+		std::string extension = (problem_path.size() > 5)? problem_path.substr(problem_path.find_last_of('.')) : "";
+		KCL_rosplan::ProblemGeneratorFactory::ProbGen pg_type;
```

Comment:
The problem generation node can also be used to output the problem on a message topic, and not print to file, which is actually the more normal usage to connect to a planner interface. In those instances a user might well leave the problem file parameter blank. The default behaviour according to this code is to use the PDDLGenerator, which I think is reasonable. 

However, an RDDL user might be confused as to how to specify that they want an RDDL problem file. Changing the problem path is an unusual switch.

I think we need to do one of three things:
(1) include an info message, i.e.: "xxx is problem path, so using PDDL problem generation."
(2) include a new parameter to choose the type (PDDL/RDDL)
(3) have two different executables (one for each type)

I would prefer either (2) or (3). I think (2) would mean less refactoring work, while (3) matches more closely the way the planner interface and dispatcher are set up. What do you think?

Reply:
I prefer the (2) thinking as a user. It's easier to change an arg or param in the launch file than changing the executable name.